modules/yieldbotBidAdapter.js

```diff
@@ -0,0 +1,614 @@
+    const url = pIdx !== -1 ? document.location.protocol + prefix.substr(pIdx + 1) : null;
+    let cookieValue = url || this.getCookie(cookieName);
+    if (!cookieValue) {
+      cookieValue = '//i.yldbt.com/m/';
+    }
+    this.setCookie(cookieName, cookieValue, this._sessionTimeout, '/');
+    return cookieValue;
+  },
+
+  /**
+   * Bidder identifier code.
+   * @type {string}
+   * @constant
+   * @memberof module:YieldbotBidAdapter
+   */
+  get code() { return 'yieldbot'; },
+
+  /**
+   * A list of aliases which should also resolve to this bidder.
+   * @type {string[]}
+   * @constant
+   * @memberof module:YieldbotBidAdapter
+   */
+  get aliases() { return []; },
+
+  /**
+   * @property {MediaType[]} [supportedMediaTypes]: A list of Media Types which the adapter supports.
+   * @constant
+   * @memberof module:YieldbotBidAdapter
+   */
+  get supportedMediaTypes() { return ['banner']; },
+
+  /**
+   * Determines whether or not the given bid request is valid.
+   *
+   * @param {BidRequest} bid The bid params to validate.
+   * @returns {boolean} True if this is a valid bid, and false otherwise.
+   * @memberof module:YieldbotBidAdapter
+   */
+  isBidRequestValid: function(bid) {
+    let invalidSizeArray = false;
+    if (utils.isArray(bid.sizes)) {
+      const arr = bid.sizes.reduce((acc, cur) => acc.concat(cur), []).filter((item) => {
+        return !utils.isNumber(item);
+      });
+      invalidSizeArray = arr.length !== 0;
+    }
+    const ret = bid &&
+            bid.params &&
+            utils.isStr(bid.params.psn) &&
+            utils.isStr(bid.params.slot) &&
+            !invalidSizeArray &&
+            !!bid.params.psn &&
+      !!bid.params.slot;
+    return ret;
+  },
+
+  /**
+   * Make a server request from the list of BidRequests.
+   *
+   * @param {BidRequest[]} bidRequests A non-empty list of bid requests which should be sent to the Server.
+   * @return ServerRequest Info describing the request to the server.
+   * @memberof module:YieldbotBidAdapter
+   */
+  buildRequests: function(bidRequests) {
+    const requests = [];
+    if (!this._optOut && !this.sessionBlocked) {
+      const searchParams = this.initBidRequestParams();
+      this._bidRequestCount++;
+
+      const pageviewIdToMap = searchParams['pvi'];
+
+      const yieldbotSlotParams = this.getSlotRequestParams(pageviewIdToMap, bidRequests);
+
+      searchParams['sn'] =
+        yieldbotSlotParams['sn'] || '';
+
+      searchParams['ssz'] =
+        yieldbotSlotParams['ssz'] || '';
+
+      const bidUrl = this.urlPrefix() +
+              yieldbotSlotParams.psn +
+              '/v1/init';
+
+      searchParams['cts_ini'] = Date.now();
+      requests.push({
+        method: 'GET',
+        url: bidUrl,
+        data: searchParams,
+        yieldbotSlotParams: yieldbotSlotParams,
+        options: {
+          withCredentials: true,
+          customHeaders: {
+            Accept: 'application/json'
```

Comment:
is this required by your endpoint? this will cause an additional preflight request for CORS.

Reply:
A request header of `Content-Type: application/json` will certainly invoke a CORS preflight OPTIONS request.

However, the `Accept: application/json` header will not. `Accept` request header is safe listed. A response header of `Content-Type: application/json` will be accepted.

At this stage the `Accept: application/json` request header is required; however, it's possible that we can also imply a JSON response from `Accept: text/plain`. For now I believe we can proceed as defined.

Hmm, will test once more for further good measure and confirm again that our friendly yet, perennial antagonist, mobile Safari does not constrain `Accept:` header values.

Reply:
Re-confirmed, `Accept: application/json` does not trigger a CORS preflight OPTIONS request.

 - iOS 11, mobile Safari
 - Android M, Chrome

Reply:
ah thanks, good catch.